core/src/p2p.rs

```diff
@@ -0,0 +1,15 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Copy)]
+pub struct P2pNetworkPubsubMessageCacheId {
+    pub source: libp2p_identity::PeerId,
+    pub seqno: u64,
+}
+
+impl P2pNetworkPubsubMessageCacheId {
```

Comment:
@0xMimir does this belong here? I seems it belongs to the `p2p` crate, since this is p2p specific, and not part of `core`. I would move this there and get rid of the `libp2p-identity` dependency from `core`.